options/m_option.h

```diff
@@ -633,6 +632,7 @@ extern const char m_option_path_separator;
 #define OPTDEF_DOUBLE(d)  .defval = (void *)&(const double){d}
 
 #define M_RANGE(a, b) .min = (double) (a), .max = (double) (b)
+#define M_DEPRECATED .deprecation_message = ""
```

Comment:
Could you clarify if `.deprecation_message` is suppose to still be used, or we defer fully to `interface-changes.rst`? In which case we could simplify it and `.flags = M_OPT_DEPRECATED` for example.

Reply:
> Replaced with a boolean flag instead.

why not use existing `.flags` field?

Reply:
I don't see the appeal of `M_DEPRECATED` instead of doing `.flags = M_OPT_DEPRECATED` or possibly `.flags = M_OPT_DEPRECATED | M_OPT_DEFAULT_NAN` and so on. Yes, you have to or it, just like in all other cases of usage of flags. But fine if you need to have dedicated field for deprecated flag.